lib/presentation/mixins/media_picker_mixin.dart

```diff
@@ -1,92 +1,71 @@
+    as linagora_image_picker;
 import 'package:linagora_design_flutter/images_picker/images_picker_grid.dart';
 import 'package:linagora_design_flutter/images_picker/use_camera_widget.dart';
 import 'package:matrix/matrix.dart';
 import 'package:permission_handler/permission_handler.dart';
-import 'package:wechat_camera_picker/wechat_camera_picker.dart';
+import 'package:photo_manager/photo_manager.dart';
 
-typedef OnSendTap = void Function()?;
+import 'common_media_picker_mixin.dart';
 
-mixin ImagePickerMixin {
-  final ImagePickerGridController imagePickerController =
-      ImagePickerGridController();
+typedef OnCameraPicked = void Function(AssetEntity assetEntity)?;
 
-  final numberSelectedImagesNotifier = ValueNotifier<int>(0);
-
-  List<ChatActions> get listChatActions => [
-        ChatActions.gallery,
-        ChatActions.documents,
-        ChatActions.location,
-        ChatActions.contact
+mixin MediaPickerMixin on CommonMediaPickerMixin {
+  List<PickerType> get listChatActions => [
+        PickerType.gallery,
+        PickerType.documents,
+        PickerType.location,
+        PickerType.contact
       ];
 
-  void listenToSelectionInImagePicker() {
-    imagePickerController.addListener(() {
-      numberSelectedImagesNotifier.value =
-          imagePickerController.selectedAssets.length;
-    });
-  }
-
-  void removeAllImageSelected() {
-    imagePickerController.clearAssetCounter();
-    numberSelectedImagesNotifier.value = 0;
-  }
-
-  Future<PermissionStatus>? getCurrentPhotoPermission() {
-    return PermissionHandlerService().requestPermissionForPhotoActions();
-  }
-
-  Future<PermissionStatus>? getCurrentCameraPermission() {
-    return PermissionHandlerService().requestPermissionForCameraActions();
-  }
-
-  void showImagesPickerBottomSheetAction({
+  void showMediasPickerBottomSheetAction({
     required BuildContext context,
-    required OnItemAction onItemAction,
+    required ImagePickerGridController imagePickerGridController,
+    OnPickerTypeTap? onPickerTypeTap,
     Room? room,
-    OnSendTap onSendTap,
+    OnSendPhotosTap onSendTap,
+    OnCameraPicked? onCameraPicked,
   }) async {
     final currentPermissionPhotos = await getCurrentPhotoPermission();
     final currentPermissionCamera = await getCurrentCameraPermission();
     if (currentPermissionPhotos != null && currentPermissionCamera != null) {
-      showImagesPickerBottomSheet(
+      showMediasPickerBottomSheet(
         context: context,
+        imagePickerController: imagePickerGridController,
         permissionStatusPhotos: currentPermissionPhotos,
         permissionStatusCamera: currentPermissionCamera,
         onSendTap: onSendTap,
         room: room,
-        onItemAction: onItemAction,
-      ).whenComplete(() => removeAllImageSelected());
+        onPickerTypeTap: onPickerTypeTap,
+      );
     }
   }
 
-  Future<void> showImagesPickerBottomSheet({
+  Future<void> showMediasPickerBottomSheet({
```

Comment:
```suggestion
  Future<void> showMediaPickerBottomSheet({
```
medias is unknown word